vscode-lean4/src/utils/leanInstaller.ts

```diff
@@ -463,23 +475,33 @@ export class LeanInstaller implements Disposable {
         } else {
             const terminalName = 'Lean installation via elan';
 
+            let executionPolicy = '';
             let terminalOptions: TerminalOptions = { name: terminalName };
             if (process.platform === 'win32') {
-                const windir = process.env.windir
-                terminalOptions = { name: terminalName, shellPath: `${windir}\\System32\\WindowsPowerShell\\v1.0\\powershell.exe` };
+                terminalOptions = { name: terminalName, shellPath: this.getPowerShell() };
             }
             const terminal = window.createTerminal(terminalOptions);
             terminal.show();
 
+            const thisObject = this;
             // We register a listener, to restart the Lean extension once elan has finished.
             const result = new Promise<boolean>(function(resolve, reject) {
-                window.onDidCloseTerminal((t) => {
+                window.onDidCloseTerminal(async (t) => {
                 if (t.name === terminalName) {
+                    if (executionPolicy === 'Restricted'){
+                        console.log('Restoring ExecutionPolicy to Restricted')
+                        await thisObject.setExecutionPolicy(executionPolicy);
+                    }
                     resolve(true);
                 }});
             });
 
             if (process.platform === 'win32') {
+                executionPolicy = (await this.getExecutionPolicy()).trim();
+                if (executionPolicy === 'Restricted'){
+                    console.log('Setting ExecutionPolicy to RemoteSigned so we can run elan-init.ps1')
+                    await this.setExecutionPolicy('RemoteSigned');
+                }
                 terminal.sendText(
                     `Invoke-WebRequest -Uri "${this.leanInstallerWindows}" -OutFile elan-init.ps1\n` +
```

Comment:
Can't you just run `Set-ExecutionPolicy -ExecutionPolicy Unrestricted -Scope Process` in this script?

Reply:
I suppose the goal is to make sure the script in question is properly signed, which is more secure against the (admittedly unlikely) case that the downloaded `elan-init.ps1` was corrupted.

Reply:
@Kha , you are right, that is a much better solution, is now pushed.
@tydeu , yes, since the scope is now Process, and this will be limited to the terminal window we open to install elan and the terminal is closed after the install finishes, the only real attack vector there is elan-init.ps1 itself, so yes.

Reply:
I mostly used `Unrestricted` because `RemoteSigned` kind of implies that `elan-init.ps1` is signed, which it's not. It feels like an unnecessary gotcha that `Invoke-WebRequest` does not mark the downloaded script as a remote file.

Reply:
RemoteSigned is safer so I'd like to leave it there.  Nothing we can do about the fact that powershell chose some weird names.  The theory is that if the app has permission to do Invoke-WebRequest to download something locally, then the app already has enough trust to run these scripts, but as Mac pointed out, RemoteSigned  still guards against that script doing something unexpected.

Reply:
Never mind, I decided to switch to Unrestricted because I found other problems with RemoteSigned.